store/etcd/rulesets.go

```diff
@@ -89,6 +91,68 @@ func (s *RulesetService) List(ctx context.Context, prefix string, limit int, con
 	return &entries, nil
 }
 
+// ListPaths returns all rulesets paths under the given prefix.
+// If the prefix is empty, it returns **all** the rulesets paths.
+func (s *RulesetService) ListPaths(ctx context.Context, prefix string, limit int, continueToken string) (*store.RulesetEntries, error) {
+	options := make([]clientv3.OpOption, 0, 3)
+
+	var key string
+
+	if limit < 0 || limit > 100 {
+		limit = 50 // TODO(asdine): make this configurable in future releases.
+	}
+
+	if continueToken != "" {
+		lastPath, err := base64.URLEncoding.DecodeString(continueToken)
+		if err != nil {
+			return nil, store.ErrInvalidContinueToken
+		}
+
+		key = string(lastPath)
+
+		rangeEnd := clientv3.GetPrefixRangeEnd(s.latestRulesetPath(prefix))
+
+		options = append(options, clientv3.WithRange(rangeEnd))
+	} else {
+		key = prefix
+		options = append(options, clientv3.WithPrefix())
+	}
+
+	options = append(options, clientv3.WithKeysOnly())
+	options = append(options, clientv3.WithLimit(int64(limit)))
+
+	resp, err := s.Client.KV.Get(ctx, s.latestRulesetPath(key), options...)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to fetch all entries")
+	}
+
+	// if a prefix is provided it must always return results
+	// otherwise it doesn't exist.
+	if resp.Count == 0 && prefix != "" {
+		return nil, store.ErrNotFound
+	}
+
+	var entries store.RulesetEntries
+	entries.Revision = strconv.FormatInt(resp.Header.Revision, 10)
+	for _, pair := range resp.Kvs {
+		sk := strings.Split(string(pair.Key), "/")
```

Comment:
📖 This is not the best / simplest way to do that. Can you find a simpler way to remove the namespace from the key?

Reply:
Done in https://github.com/heetch/regula/pull/43/commits/f4c3b4cb428e0e51f9dd5ef5f0685a68cefbb540